addons/music/fnc_compileMusic.sqf

```diff
@@ -1,40 +1,54 @@
 /* ----------------------------------------------------------------------------
 Function: CBA_fnc_compileMusic
+
 Description:
-A function used to gather a list of all usable music.
+    A function used to gather a list of all usable music.
+    
 Parameters:
-none
-Example:
-(begin example)
-_allmusic = [] call CBA_fnc_compileMusic
-(end example)
+    none
+
 Returns:
-config
+    Array of compiled music
+    
+Example:
+    (begin example)
+        _allMusic = [] call CBA_fnc_compileMusic
+    (end example)
+
 Author:
-Fishy
+    Fishy
 ---------------------------------------------------------------------------- */
 #include "script_component.hpp"
+
 if (isNil QGVARMAIN(compiledMusic)) then {
-_config = configFile >> 'CfgMusic';
-_missionConfig = missionConfigFile >> 'CfgMusic';
-_unsortedSongs = [];
-for [{_i=0}, {_i < (count _config)}, {INC(_i)}] do {
-_song = _config select _i;
-_songDuration = getNumber (_song >> 'duration');
-if (_songDuration !=0) then {
-_songClassName = configName _song;
-_unsortedSongs pushBack _songClassName;
-};
-};
-for [{_i=0}, {_i < (count _missionConfig)}, {INC(_i)}] do {
-_song = _missionConfig select _i;
-_songDuration = getNumber (_song >> 'duration');
-if (_songDuration !=0) then {
-_songClassName = configName _song;
-_unsortedSongs pushBack _songClassName;
-};
-};
-GVARMAIN(compiledMusic) = _unsortedSongs;
-_unsortedSongs 
+    private "_config";
+    _config = configFile >> 'CfgMusic';
+    private "_missionConfig";
+    _missionConfig = missionConfigFile >> 'CfgMusic';
+    private "_unsortedSongs";
+    _unsortedSongs = [];
+    
+    for [{_i=0}, {_i < (count _config)}, {INC(_i)}] do {
```

Comment:
you are only using _i to select from _missionConfig. So a forEach should work here as well shouldn't it?
Also please don't `private "_var";_var = 1;` do `private _var = 1;` instead